### ISSUE TITLE
Allow more sane polling frequency choices

### ISSUE THREAD

**Issue by skx** [skx/rss2email#133]
When this application was released there was a hardcoded behaviour:

* Process every feed known.
* Sleep 15 minutes.
* Restart

Later it became possible to override the 15-minute delay, to make it longer/shorter.  Now that we have caching I don't feel too bad about polling every 30 minutes, but it has to be said that we're probably not being well-behaved.

The specific problem I have is that we cannot say:

* Poll example.com, and example.net every 15 minutes.
* BUT poll example.org every 50 minutes.

I have the feeling this is something that _should_ be supported.  We could make some feed skip if we knew our frequency and expected number of times to poll.  Or we could count sequential runs so we pretend we have 24 runs and a feed could say:

-  run: 1,10

Or 

- run: hr1   hr4

But this kinda relies on some assumptions and it starts to feel complex.  I don't want to start storing "arbitrary state" for feeds, but I do want to too.  Help?

I know we have the cache, and that could be used as a proxy.  If we assume we're invoked and running "frequently" we could add a timestamp to the URL-cache and then say:

* refetch: 2hr

That way if the cached entry was >2hrs we'd poll the feed again  (note here I'm using the age of the cache entry as a "last time this feed was fetched", I'm not talking about a full-request vs a cached request being sent.  I'm using it as implicit state that goes away upon restart - on the assumption that we're running under deamon mode.)

That might work, but I need to think about it.

TLDR:  The default is fine, but I think we should be able to change the frequency on a per-feed basis and a global "sleep then repeat" setting is not the right approach.  Some feeds track conversation (reddit, blog-comments, hackernews) and need to be "fast".  Other feeds describe youtube videos and only update once every two days at the most frequenty.

**Comment by StayPirate** [comment]
Nice idea!

_Brainstorm mode on_

What if you change the default behavior for the daemon mode and make it always polling, as it loops on feed.txt and eliminating `sleep-delay`. Then you implicit consider each feed that do not specify a custom `frequency` value as `frequency = 15m`.

That way even if I set `frequency = 5m` on a specific feed (I think nobody would do that) that will be honored. Supposing a full loop cycle takes less then 5m.

_Brainstorm mode off_

TBH I don't understand what you mean with `run: 1,10` and `run: hr1 hr4`. And yeah, I would say that it does seem too complex. But the new option you proposed in #134 is pretty easy to understand and I like it.

**Comment by skx** [comment]
For the "`run: 1,10`", or "`run: hr1 hr4`", I had the idea of dividing the day into five-minute buckets, using the current hour/minute.   Then deciding on a per-feed basis which bucket would have a given run.     Not sure if that makes it clearer, or not, but the current idea won for simplicity.

Eliminating sleep delay seems like it might work, and avoids the confusion of what happens when the sleep-delay is longer than the frequency - thank you!
